virtualizarr/tests/test_writers/test_icechunk_append.py

```diff
@@ -0,0 +1,333 @@
+
+
+def generate_chunk_manifest(
+    netcdf4_file: str,
+    shape: tuple[int, int] = (3, 4),
+    chunks: tuple[int, int] = (3, 4),
+    base_offset=6144,
+    length=48,
+) -> ChunkManifest:
+    chunk_dict = {}
+    num_chunks_x = shape[0] // chunks[0]
+    num_chunks_y = shape[1] // chunks[1]
+    if len(shape) == 3:
+        num_chunks_z = shape[2] // chunks[2]
+    offset = base_offset
+
+    for i in range(num_chunks_x):
+        for j in range(num_chunks_y):
+            if len(shape) == 3:
+                for k in range(num_chunks_z):
+                    chunk_index = f"{i}.{j}.{k}"
+                    chunk_dict[chunk_index] = {
+                        "path": netcdf4_file,
+                        "offset": offset,
+                        "length": length,
+                    }
+                    offset += length
+            else:
+                chunk_index = f"{i}.{j}"
+                chunk_dict[chunk_index] = {
+                    "path": netcdf4_file,
+                    "offset": offset,
+                    "length": length,
+                }
+                offset += length  # Increase offset for each chunk
+    return ChunkManifest(chunk_dict)
+
+
+def gen_virtual_dataset(
+    file_uri: str,
+    shape: tuple[int, int] = (3, 4),
+    chunk_shape: tuple[int, int] = (3, 4),
+    dtype: np.dtype = np.dtype("int32"),
+    compressor: dict = None,
+    filters: str = None,
+    fill_value: str = None,
+    encoding: dict = None,
+    variable_name: str = "foo",
+    base_offset: int = 6144,
+    length: int = 48,
+    dims: list[str] = None,
+):
+    manifest = generate_chunk_manifest(
+        file_uri,
+        shape=shape,
+        chunks=chunk_shape,
+        base_offset=base_offset,
+        length=length,
+    )
+    zarray = ZArray(
+        shape=shape,
+        chunks=chunk_shape,
+        dtype=dtype,
+        compressor=compressor,
+        filters=filters,
+        fill_value=fill_value,
+    )
+    ma = ManifestArray(chunkmanifest=manifest, zarray=zarray)
+    ds = open_dataset(file_uri)
+    dims = dims or ds.sizes.keys()
+    var = Variable(
+        data=ma,
+        dims=dims,
+        encoding=encoding,
+        attrs=ds[variable_name].attrs,
+    )
+    return Dataset(
+        {variable_name: var},
+    )
+
+
+# Success cases
+
+
+## When appending to a single virtual ref without encoding, it succeeds
+def test_append_virtual_ref_without_encoding(
+    icechunk_storage: "StorageConfig", simple_netcdf4: str
+):
+    import xarray as xr
+    from icechunk import IcechunkStore
+
+    # generate virtual dataset
+    vds = gen_virtual_dataset(file_uri=simple_netcdf4)
+
+    # create the icechunk store and commit the first virtual dataset
+    icechunk_filestore = IcechunkStore.create(storage=icechunk_storage)
+    dataset_to_icechunk(vds, icechunk_filestore)
+    icechunk_filestore.commit(
+        "test commit"
+    )  # need to commit it in order to append to it in the next lines
```

Comment:
We need to open the existing store in append mode in order to append otherwise I get the error:

```
zarr.errors.ContainsGroupError: A group exists in store <icechunk.IcechunkStore object at 0x10eaf9100> at path ''.
```

That's the error that I get just trying to use the store object from `IcechunkStore.create(`. But if I do use a store with mode='a' but _do not_ commit to the first store object, I get the following error:

```
FileNotFoundError: <icechunk.IcechunkStore object at 0x10960d490>
```